api/src/test/java/org/openmrs/module/fhir2/api/search/AllergyIntoleranceSearchQueryImplTest.java

```diff
@@ -490,4 +494,86 @@ public void searchForAllergies_shouldHandleComplexQuery() {
 		    equalTo(PATIENT_UUID));
 	}
 	
+	@Test
+	public void shouldReturnCollectionOfAllergiesSortedBySeveritySevere() {
+		
+		SortSpec sort = new SortSpec();
+		sort.setParamName(AllergyIntolerance.SP_SEVERITY);
+		sort.setOrder(SortOrderEnum.ASC);
+		
+		IBundleProvider results = search(
+		 new SearchParameterMap().addParameter(FhirConstants.SEVERITY_SEARCH_HANDLER, FhirConstants.GLOBAL_PROPERTY_SEVERE, new TokenAndListParam().addAnd(new TokenParam(SEVERITY_SEVERE))).setSortSpec(sort));
+		
+		sort.setOrder(SortOrderEnum.DESC);
+		for (int i = 1; i < get(results).size(); i++) {
+			
+			assertThat(((AllergyIntolerance) get(results).get(i - 1)).getReactionFirstRep().getSeverity(),
+				    lessThanOrEqualTo(((AllergyIntolerance) get(results).get(i)).getReactionFirstRep().getSeverity()));	
+		}
```

Comment:
@gitcliff This should be something like:

```suggestion
		List<AllergyIntolerance> fullResults = (List<AllergyIntolerance>) get(results);
		for (int i = 1; i < fullResults.size(); i++) {
			assertThat(fullResults.get(i - 1).getReactionFirstRep().getSeverity(),
				    lessThanOrEqualTo(fullResults.get(i).getReactionFirstRep().getSeverity()));	
		}
```

Reply:
Similar comments apply to the rest of the sorting test methods.

Reply:
@ibacher  the above change you suggested which involves casting drops `Cannot cast from List<IBaseResource> to List<AllergyIntolerance>` compilation error

Reply:
You can do it using the following approach- 
`get(results).stream().map(p -> (AllergyIntolerance)p).collect(Collectors.toList());`

Reply:
thanks @varung-31  let me try it out